src/wallet/wallet.cpp

```diff
@@ -1442,9 +1443,12 @@ void CWallet::BlockDisconnected(const CBlock& block, int height)
     auto locked_chain = chain().lock();
     LOCK(cs_wallet);
 
+    m_last_block_processed_height = height - 1;
     for (const CTransactionRef& ptx : block.vtx) {
         SyncTransaction(ptx, {} /* block hash */, 0 /* position in block */);
     }
+
+    m_last_block_processed = block.hashPrevBlock;
```

Comment:
3d32a691d2df4c73a39a64a5d331c29b5eb94213

Is this a bugfix?

Reply:
Yes it was a bugfix compare to a53586c at least, it become obvious only after switching to query the chain based on block hash at wallet loading!